File: WinForms/DevExpress.WindowsMailClient.Win/CS/DevExpress.WindowsMailClient.Win/Modules/Scheduler/SchedulerModule.cs
using System;
using DevExpress.Utils.Drawing.Helpers;
using DevExpress.WindowsMailClient.Win.ViewModels;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Demos.DemoData;

namespace DevExpress.WindowsMailClient.Win.Modules {
    public partial class SchedulerModule : BaseModule {
        public SchedulerModule() : base(typeof(SchedulerModuleViewModel)) {
            InitializeComponent();
            this.schedulerControl1.BeginUpdate();
            try
            {
                this.schedulerControl1.OptionsBehavior.UseAsyncMode = false;
                this.schedulerControl1.MonthView.AllowScrollAnimation = false;
            }
            finally {
                this.schedulerControl1.EndUpdate();
            }
        }
        public SchedulerControl Scheduler { get { return this.schedulerControl1; } }

        void SchedulerModule_Load(object sender, EventArgs e) {
            Scheduler.Start = SchedulerDataHelper.Start;
            SchedulerDataHelper.Attach(this.schedulerControl1);
        }
        protected override void WndProc(ref System.Windows.Forms.Message m) {
            if(m.Msg == 528 || m.Msg == MSG.WM_LBUTTONDBLCLK || m.Msg == MSG.WM_LBUTTONDOWN || m.Msg == MSG.WM_LBUTTONDOWN) { 
                this.GetParentViewModel<MainViewModel>().HideAllPanels();
            }
            base.WndProc(ref m);
        }
    }
}

File: WinForms/DevExpress.WindowsMailClient.Win/VB/DevExpress.WindowsMailClient.Win/Modules/Scheduler/SchedulerModule.vb
Imports System
Imports DevExpress.Utils.Drawing.Helpers
Imports DevExpress.WindowsMailClient.Win.ViewModels
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Demos.DemoData

Namespace DevExpress.WindowsMailClient.Win.Modules

    Public Partial Class SchedulerModule
        Inherits BaseModule

        Public Sub New()
            MyBase.New(GetType(SchedulerModuleViewModel))
            InitializeComponent()
            schedulerControl1.BeginUpdate()
            Try
                schedulerControl1.OptionsBehavior.UseAsyncMode = False
                schedulerControl1.MonthView.AllowScrollAnimation = False
            Finally
                schedulerControl1.EndUpdate()
            End Try
        End Sub

        Public ReadOnly Property Scheduler As SchedulerControl
            Get
                Return schedulerControl1
            End Get
        End Property

        Private Sub SchedulerModule_Load(ByVal sender As Object, ByVal e As EventArgs)
            Scheduler.Start = SchedulerDataHelper.Start
            SchedulerDataHelper.Attach(schedulerControl1)
        End Sub

        Protected Overrides Sub WndProc(ByRef m As Windows.Forms.Message)
            If m.Msg = 528 OrElse m.Msg = MSG.WM_LBUTTONDBLCLK OrElse m.Msg = MSG.WM_LBUTTONDOWN OrElse m.Msg = MSG.WM_LBUTTONDOWN Then
                Me.GetParentViewModel(Of MainViewModel)().HideAllPanels()
            End If

            MyBase.WndProc(m)
        End Sub
    End Class
End Namespace
